components/post_view/post_header/post_header.tsx

```diff
@@ -67,17 +67,17 @@ export type Props = {
     /**
      * Whether or not the post username can be overridden.
      */
-    enablePostUsernameOverride: boolean;
+    enablePostUsernameOverride?: boolean;
```

Comment:
post_view/post/post.tsx was a JS component and didn't provide these bools, effectively defaulting to false, but when I converted I got type errors. Not sure if they *should* be false when rendered there, but to not change types or behavior I can add explicit `false` values. If not passing them at all was an error, we can look at passing the correct values, instead.

Reply:
Oh, I see what's happening. Those props come from redux, but because we use this `Props` type for `ownProps` in this component's [`mapStateToProps`](https://github.com/mattermost/mattermost-webapp/blob/master/components/post_view/post_header/index.ts#L16), React expects them to be provided from outside.

Instead of passing these in from the Post component, could you instead fix the type of `ownProps` in the PostHeader's `mapStateToProps` so that it only includes the props passed from outside the component (the post)?